ModulemdTranslationHelpers/Utils.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+import gi
+gi.require_version('Modulemd', '2.0')
+from gi.repository import Modulemd
+
+from babel.messages import Catalog, pofile
+from babel._compat import StringIO, BytesIO
+from datetime import datetime
+
+# Create a Modulemd.ModuleIndex object containing the YAML stream
+fedora_index = Modulemd.ModuleIndex.new()
+fedora_yaml_path = "../modulemd/v2/tests/test_data/f29.yaml"
+ret, failures = fedora_index.update_from_file(fedora_yaml_path, True)
+
+# Retrieve all Modulemd.Module objects
+module_names = fedora_index.get_module_names()
+
+# Create a dictionary where
+# key: stream name
+# value: NSVCA of highest version of that stream
+final_streams = {}
+
+for mod_name in module_names:
+    module = fedora_index.get_module(mod_name)
+
+    # Retrieve all Modulemd.ModuleStreamV2 objects from module
+    streams = module.get_all_streams()
+    for stream in streams:
+        key = stream.get_stream_name()
+        if key not in final_streams.keys():
+            final_streams[key] = stream.get_NSVCA()
+        else:
+            nsvca = stream.get_NSVCA()
+            version = int(nsvca.split(':')[2])
+            if version > int(final_streams[key].split(
+                    ":")[2]):  # found a higher version for same stream
+                final_streams[key] = nsvca
+
+summary_dict = {}
+description_dict = {}
+profile_desc_default_dict = {}
+profile_desc_minimal_dict = {}
+
+for stream_name, nsvca in final_streams.items():
+    mdname = nsvca.split(":")[0]
+    version = int(nsvca.split(":")[2])
+    context = nsvca.split(":")[3]
+    arch = nsvca.split(":")[4]
+    module = fedora_index.get_module(mdname)
+    stream = module.get_stream_by_NSVCA(stream_name, version, context, arch)
+
+    # Process description
+    description = stream.get_description("C")
+    if description not in description_dict.keys():
+        description_dict[description] = [nsvca.rsplit(
+            ":", 3)[0] + ";description"]  # name:stream
+    else:
+        description_dict[description].append(
+            nsvca.rsplit(":", 3)[0] + ";description")
```

Comment:
You're getting too fancy with the `rsplit()` and concatenation here. Just do

```suggestion
        description_dict[description].append("{}:{};description" % (stream.props.module_name, stream.props.stream_name))
```